src/steiner_tree.jl

```diff
@@ -9,5 +9,13 @@ using SimpleTraits: SimpleTraits, Not, @traitfn
     map(v -> vertex_positions(g)[v], term_vert),
     dist_matrix_to_position_dist_matrix(g, distmx),
   )
-  return typeof(g)(position_tree, map(v -> ordered_vertices(g)[v], vertices(position_tree)))
+  named_st = typeof(g)(
+    position_tree, map(v -> ordered_vertices(g)[v], vertices(position_tree))
+  )
+  # Detect and remove vertices of degree zero
+  zero_verts = filter(v -> degree(named_st, v) == 0, vertices(named_st))
+  for v in zero_verts
+    rem_vertex!(named_st, v)
+  end
```

Comment:
```suggestion
  for v in vertices(tree)
    iszero(degree(tree, v)) && rem_vertex!(tree, v)
  end
```
Avoids the intermediate data structure and a bit more readable.

Reply:
I was worried that mixing the two steps this way would be dangerous, since I originally tried a similar pattern on a `SimpleGraph` and it failed badly since vertex labels kept getting rewritten.

But here I suppose it's totally safe since the vertex names are unique.

Reply:
I see, I don't anticipate an issue like that here since as you say the vertex names are unique.

Reply:
Unfortunately the above version does lead to a bug. The reason is that `vertices(tree)` returns a reference, not a copy, to the vertices, so as `rem_vertex!` acts, it reduces the value of `nv(tree)` on the fly and the iteration ends too early. One fix that I tested could be:
```
  verts = copy(vertices(tree))
  for v in verts
    iszero(degree(tree, v)) && rem_vertex!(tree,v)
  end
```

Reply:
Also this works:
```
  for v in copy(vertices(tree))
    iszero(degree(tree, v)) && rem_vertex!(tree,v)
  end
```

Reply:
I was wondering about that, that sounds like a good fix.